src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -12483,6 +12823,52 @@ public void SynthesizedDelegateTypes_DefaultParameterValues_StringUnification()
                 """).VerifyDiagnostics();
         }
 
+        [Fact]
+        public void SynthesizedDelegateTypes_DefaultParameterValues_LeastOverriddenMethod()
+        {
+            var source = """
+                var d = new D().M;
+                System.Console.WriteLine(d.GetType());
+
+                public class D : C
+                {
+                    public override void M(int x) { }
+                }
+
+                public abstract class C
+                {
+                    public abstract void M(int x = 42);
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,9): error CS8917: The delegate type could not be inferred.
```

Comment:
>// (1,9): error CS8917: The delegate type could not be inferred. [](http://example.com/codeflow?start=16&length=64)

This error feels unexpected. Overrides shouldn't be causing ambiguities o their own. I think overload resolution picks up constant values from the most derived declarations

Reply:
In order to properly deal with default values, I think we actually need to drop less derived methods from the set of methods that will be used to infer delegate's signature. Probably do something similar to what ```OverloadResolution.RemoveLessDerivedMembers``` is doing . And we probably should test every unique scenario that helper is handling. It looks like it applies filtering for scenarios that do not involve regular overriding.

Reply:
I've pushed changes that seem to do what I want, but don't use  RemoveLessDerivedMembers. I need to investigate more to understand what that's doing and how is it different from MemberGroupContainsMoreDerivedOverride